### PR TITLE
refactor: port UserProfile to TS/SWR

### PR DESCRIPTION
**Pull request by olav**

### CODE REVIEW

File: src/component/user/UserProfile/UserProfile.tsx [hunk 1]
```diff
@@ -8,15 +8,20 @@ import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
 import { useStyles } from './UserProfile.styles';
 import { useCommonStyles } from '../../../common.styles';
 import UserProfileContent from './UserProfileContent/UserProfileContent';
+import { IUser } from "../../../interfaces/user";
+
+interface UserProfileProps {
```
Comment (FredrikOseberg) [inline src/component/user/UserProfile/UserProfile.tsx:13]:
```suggestion
interface IUserProfileProps {
```

File: src/component/user/UserProfile/UserProfileContent/UserProfileContent.tsx [hunk 2]
```diff
@@ -17,6 +17,17 @@ import EditProfile from '../EditProfile/EditProfile';
 import legacyStyles from '../../user.module.scss';
 import { getBasePath } from '../../../../utils/format-path';
 import useUiConfig from '../../../../hooks/api/getters/useUiConfig/useUiConfig';
+import { IUser } from "../../../../interfaces/user";
+
+interface UserProfileContentProps {
```
Comment (FredrikOseberg) [inline src/component/user/UserProfile/UserProfileContent/UserProfileContent.tsx:22]:
```suggestion
interface IUserProfileContentProps {
```

File: src/component/user/UserProfile/UserProfile.tsx [hunk 3]
```diff
@@ -8,15 +8,20 @@ import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
...
+    updateSettingLocation: (field: 'locale', value: string) => void
+}
 
 const UserProfile = ({
     profile,
     location,
-    fetchUser,
     updateSettingLocation,
-}) => {
+}: UserProfileProps) => {
```
Comment (FredrikOseberg) [inline src/component/user/UserProfile/UserProfile.tsx:22]:
```suggestion
}: IUserProfileProps) => {
```

File: src/component/user/UserProfile/UserProfileContent/UserProfileContent.tsx [hunk 4]
```diff
@@ -26,27 +37,31 @@ const UserProfileContent = ({
     imageUrl,
     currentLocale,
     setCurrentLocale,
-}) => {
+}: UserProfileContentProps) => {
```
Comment (FredrikOseberg) [inline src/component/user/UserProfile/UserProfileContent/UserProfileContent.tsx:40]:
```suggestion
}: IUserProfileContentProps) => {
```

**Review by FredrikOseberg — APPROVED** [review]
LGTM!

File: src/component/user/UserProfile/UserProfile.tsx (hunk 1 above)
Comment (olav) [reply src/component/user/UserProfile/UserProfile.tsx]: Nice catch! One day I'll remember these.

**Pull request merged**